### DISCUSSION TITLE
Index Pages in markdown

### DISCUSSION THREAD

**Discussion by filmgal** [SilentVoid13/Templater#1001] [Templates Showcase]
I prefer having my Index pages in markdown rather than dataview so that I can publish them but also to future proof my md files (and see content in them rather than just dataview code).

So I created this template. This one is for my Food database.  You are looking at my Index page for Instant Pot. (I have some for Appetlizers, Desserts, Poultry, etc.  This page displays the files that include a wikilink to "[[Instant Pot]]", and groups them by recipes I've made, not made, and reference docs.

I use a similar template to update a few film pages I have. I include those below.
## Food
left: reading mode, right: editing mode
![image](https://user-images.githubusercontent.com/30246308/213890241-bac108c5-161b-420e-b6b0-29b5732f135f.png)

I have to manually update these index pages (for now), but it's not a huge deal. Whenever I find myself going to an index page I just select all, delete, and call my template.

### code
```

---
title: <%tp.file.title%>
aliases: <%tp.file.title%> Index
datecreated: <%tp.file.creation_date("YYYY-MM-DD")%>
datemodified: <%tp.date.now("YYYY-MM-DD")%>
---
%% to update page select all, delete, and call  [[MD Index Pages]] Template %%

# <%tp.file.title%> Index

Made

<%* const made = DataviewAPI.pages("[[" + tp.file.title + "]]" + 'and #made') 
.where(p => p.file.name !="Main Page" && !p.file.folder.includes("Admin/Templates")) 
.sort(p => p.file.name, 'asc') 
.map(p => p.file.link); 
const listMade = DataviewAPI.markdownList(made); 
-%> 
<%- listMade -%>

To Try
<%* const toTry = DataviewAPI.pages("[[" + tp.file.title + "]]" + 'and !#made')
.where(p => p.file.name !="Main Page" && p.file.folder.includes("Recipes"))
.sort(p => p.file.name, 'asc') 
.map(p => p.file.link); 
const listToTry = DataviewAPI.markdownList(toTry); 
-%> 
<%- listToTry -%>

Reference
<%* const reference = DataviewAPI.pages("[[" + tp.file.title + "]]") 
.where(p => p.file.folder.includes("Reference"))
.sort(p => p.file.name, 'asc') 
.map(p => p.file.link); 
const listReference = DataviewAPI.markdownList(reference); 
-%> 
<%- listReference -%>


```

## Film

I have a folder of lists, including one page for each of my favourite directors, to keep track of the films I've yet to see from them.
I create new file with their name (e.g., Jacques Audiard) after that I use my MD New Director Page template to create the page.  When I later need to update it, I just remove all the content after "Saw" and use an MD Update Director Page which is basically the same but starting at "Saw".

![image](https://user-images.githubusercontent.com/30246308/213895238-5e3d2406-37b6-4386-98de-7c2c59edcf6d.png)

### code
```

---
datecreated: <%tp.file.creation_date("YYYY-MM-DD")%>
aliases: "<%tp.file.title%>"
imdbsearch: https://www.imdb.com/find?q=<%tp.file.title.replace(/\s+/g, "+")%>&s=names
lbsearch: https://letterboxd.com/search/cast-crew/<%tp.file.title.replace(/\s+/g, "+")%>/
---

%%
This will only work if there is an aliases field in the director's page - so this page adds one.
To update this page later, select everything from "Saw" to the end of the file and call the [[MD Update Director Page]]
%%

# <%tp.file.title%>
[search on imdb](https://www.imdb.com/find?q=<%tp.file.title.replace(/\s+/g, "+")%>&s=names)   [search on letterboxd](https://letterboxd.com/search/cast-crew/<%tp.file.title.replace(/\s+/g, "+")%>/)

---

_last updated: \<\%\+ tp.file.last_modified_date("YYYY-MM-DD HH:mm")\%\>_

---
Saw

<%*
const FilmsSeen = DataviewAPI.pages('#filmtitle and #saw')
  .where (p => tp.frontmatter.aliases?.includes(p.director) || p.director?.includes(tp.file.title) || p.director?.includes(tp.frontmatter.aliases))
  .sort(p => p.file.name, 'asc')
  .map(k => k.file.link + "   •   " + k.genre);
const ListFilmsSeen = DataviewAPI.markdownList(FilmsSeen);
-%>
<% ListFilmsSeen -%>

---
Not Seen

<%*
const Films = DataviewAPI.pages('#filmtitle and !#saw')
 .where (p => tp.frontmatter.aliases?.includes(p.director) || p.director?.includes(tp.file.title) || p.director?.includes(tp.frontmatter.aliases))
  .sort(p => p.file.name, 'asc')
  .map(k => k.file.link + "   •   " + k.genre + "   •   " + k.runtime + " mins");
const ListFilms = DataviewAPI.markdownList(Films);
-%>
<% ListFilms -%>


---

ALL films, sorted by date, most recent first

<%*
const FilmsAll = DataviewAPI.pages('#filmtitle')
  .where (p => tp.frontmatter.aliases?.includes(p.director) || p.director?.includes(tp.file.title) || p.director?.includes(tp.frontmatter.aliases))
  .sort(p => p.filmyear, 'desc')
  .map(k => k.file.link + "   •   " + k.genre + "   •   " + k.runtime + " mins");
const ListFilmsAll = DataviewAPI.markdownList(FilmsAll);
-%>
<% ListFilmsAll -%>


```

**Comment by welpdx** [comment]
oh that's so cool!
I like how you use the DataviewAPI here!

**Reply by filmgal** [reply]
Thanks. My next step is to automate it, by selecting an entire folder and having each file in it replaced with the template. Or maybe auto-updating all files with "Index" or something like that.  Pretty sure it's going to take me a while to figure that out ;-)

**Comment by guihasparyk** [comment]
That's an incredible achievement here. Congrats for the code. I do prefer to acess notes trough links note from file explorer, just like an website.

I was looking way to Dynamically Index (with links written, not in dataview) my vault and notes just like TOC indexes headings inside a note. But unfortunately there is no plugin that managed to do that. If you find (or found alread) some way to do that, please share.

Example of vault arrangement:
- Folder 1
  - Note A
  - Note B
- Folder 2
  - Note C
  - Note D

Output into a unique index vault note in markdown (Vault Index.md):
- Folder 1
  - [[Note A]]
  - [[Note B]]
- Folder 2
  - [[Note C]]
  - [[Note D]]

**Comment by tbleiker** [comment]
I try to achieve something similar. It works pretty well with dataviewjs only. However, the headers generated with dataviewjs are not shown in the outline (see my post on the [obisidian forum](https://forum.obsidian.md/t/headers-generated-with-dataviewjs-not-shown-in-outline/61227)). I am playing around wiht a combination of dataviewjs and templater as well.

**Reply by guihasparyk** [reply]
I eventually dropped this project because of my low knowledge off coding and short time during weeks. I needed for very specific case of using an mobile app different from Obsidian, which I don't use anymore.

But you are doing a great job. People are contributing and sharing in the forum, that's awesome.

Good luck. 👍
